io-amqp1_0-impl/src/main/java/org/apache/pulsar/ecosystem/io/amqp/Connection.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.ecosystem.io.amqp;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Data;
+import lombok.experimental.Accessors;
+
+/**
+ * Connection with a list of URI's and whether the failover prefix should be used.
+ */
+@Data
+@Accessors(chain = true)
+public class Connection {
+
+    private boolean useFailover;
+    private List<ConnectionUri> uris;
+
+    public String getUri(){
+        String urisString = this.uris
+                .stream()
+                .map(ConnectionUri::getUri)
+                .collect(Collectors.joining(","));
+        return useFailover ? "failover:(" + urisString + ")" : urisString;
```

Comment:
I think that would be be a good idea. For our use case it does not add much, but i have made the changes as it would be better to support these.

I have committed some changes that should support these failover options. It adds a bit to the complexity of the configuration, but i hope you find it satisfactory. 

I should still change the docs to explain how to make use of the failover options.

thanks for pointing me in this direction.

Reply:
@tisonkun @gaoran10 Hopefully all your remarks ar addressed. Thanks for your review!

The docs are updated, and ci was successful. I am not fully happy with the string list base implementation both for urlOptions and the failoverConfigurationOptions. However, it suits our needs, and is in fact well beyond it ;-). 

For a next increment it could be better to provide explicit configuration options in the yaml or config instead.

Reply:
Thanks for your update! There is a holiday (-5.3) in our region so we may give feedback after a few days :D

Reply:
Great work! Thanks.